upload-api/tables/delegations.js

```diff
@@ -94,8 +101,16 @@ export function useDelegationsTable (dynamoDb, tableName, bucket) {
 
       const delegations = []
       for (const result of response.Items ?? []) {
-        const { link } = unmarshall(result)
-        delegations.push(await cidToDelegation(bucket, CID.parse(link)))
+        const { cause, link } = unmarshall(result)
+        const delegationCid = CID.parse(link)
+        if (cause) {
+          delegations.push(await findDelegationInInvocation(
+            invocationBucket, workflowBucket,
+            CID.parse(cause), delegationCid
+          ))
+        } else {
```

Comment:
ah that's an interesting idea - I'm going to keep this here for now but will explore the viability of that. 

@vasco-santos how hard would it be to backfill some invocations to avoid having to ever look at the legacy R2 bucket?

Reply:
@travis should not be difficult. We can use S3 client to read from R2 and write to S3. How are keys stored there? Only caution is to have a sha256/md5 to guarantee data integrity. If we can't get it, we need to buffer them in memory to copy (which should also be fine for the sizes they will likely have)

Reply:
ok great - will plan on that. I'm going to leave this fallback code in for now but will remove once I have a working migration script that backfills invocations